Makefile.kubebuilder.mk

```diff
@@ -0,0 +1,394 @@
+# Kubernetes API Code Generation and Manifest Management
+# Enhanced version following modern Kubernetes operator best practices
+# 
+# This file manages:
+# - Custom Resource Definition (CRD) generation
+# - Go code generation (deepcopy, client, etc.)
+# - RBAC manifest generation
+# - Webhook configuration generation
+# - Kubernetes manifest validation
+# - Code quality tools (linting, formatting, vetting)
+# - Container and build tools integration
+
+##@ Code Generation and Kubernetes Manifests
+
+# ==================================================================================
+# Configuration and Tool Versions
+# ==================================================================================
+
+# Core tool versions (aligned with kubebuilder defaults and auto-detection)
+CONTROLLER_TOOLS_VERSION ?= v0.17.3
+KUSTOMIZE_VERSION ?= v5.6.0
+# Auto-detect ENVTEST version from controller-runtime dependency
+ENVTEST_VERSION ?= $(shell go list -m -f "{{ .Version }}" sigs.k8s.io/controller-runtime 2>/dev/null | awk -F'[v.]' '{printf "release-%d.%d", $$2, $$3}')
+# Auto-detect Kubernetes version from k8s.io/api dependency  
+ENVTEST_K8S_VERSION ?= $(shell go list -m -f "{{ .Version }}" k8s.io/api 2>/dev/null | awk -F'[v.]' '{printf "1.%d", $$3}')
+
+# Code quality and development tool versions
+GOLANGCI_LINT_VERSION ?= v2.1.6
+GINKGO_VERSION ?= v2.23.4
+
+# Directories
+API_DIR := $(shell [ -d api ] && echo api || echo pkg/apis)
+CRD_DIR := config/crd/bases
+CRD_OPTIONS := config/crd
+RBAC_DIR := config/rbac
+WEBHOOK_DIR := config/webhook
+SCRIPTS_DIR := hack
+
+# Tool installation directory
+LOCALBIN ?= $(shell pwd)/bin
+
+# Tool binaries
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+KUSTOMIZE ?= $(LOCALBIN)/kustomize
+ENVTEST ?= $(LOCALBIN)/setup-envtest
+GOLANGCI_LINT ?= $(LOCALBIN)/golangci-lint
+GINKGO ?= $(LOCALBIN)/ginkgo
+
+# Generation options
+BOILERPLATE := $(SCRIPTS_DIR)/boilerplate.go.txt
+CURRENT_YEAR := $(shell date +'%Y')
+
+# Output formatting and colors - simplified for bash compatibility
+BUILD_COLOR := $(shell printf '\033[0;34m')
+GEN_COLOR := $(shell printf '\033[0;32m')
+WARN_COLOR := $(shell printf '\033[0;33m')
+NO_COLOR := $(shell printf '\033[0m')
+
+# Helper functions for output
+define log_info
+	@echo "$(GEN_COLOR)ℹ️  $(1)$(NO_COLOR)"
+endef
+
+define log_warn
+	@echo "$(WARN_COLOR)⚠️  $(1)$(NO_COLOR)"
+endef
+
+define log_build
+	@echo "$(BUILD_COLOR)🔨 $(1)$(NO_COLOR)"
+endef
```

Comment:
Log function does not work the same from everywhere.
For instance in `make update-tools` it fails for the `Downloading` logs, but works for the `Cleaning tools` log:
```
$ make update-tools
🔨 Cleaning tools
ℹ️  Tools cleaned
/bin/sh: line 1: @echo: command not found
```

Removing the `@` fixes one but sort of breaks the other:
```
$ make update-tools
echo "🔨 "Cleaning tools""
🔨 Cleaning tools
ℹ️  Tools cleaned
🔨 Downloading sigs.k8s.io/controller-tools/cmd/controller-gen@v0.17.3
```

Reply:
Do you have any suggestions here?

Reply:
> Do you have any suggestions here?

Sadly no good one :rofl:

Reply:
Works better now! :+1: